distribution/registry.go

```diff
@@ -123,9 +124,24 @@ func newRepository(
 			Class:      repoInfo.Class,
 		}
 
+		var newAuthTransport = authTransport
+		//TODO: Support multiple challenges for loading TLS configs
+		if len(realmHost) > 0 && registryResolver != nil {
+			//trying to load a new TLS config for the auth challenge
+			tlsConfig, err := registry.NewTLSConfig(realmHost, !registryResolver.IsInsecureRegistry(realmHost))
+			logrus.Debugf("Loading TLS config for host %s", realmHost)
+			if err != nil {
+				//if no TLS config was found for the auth challenge, keep the previous TLS config
+				logrus.Warnf("TLS config not found for host %s", realmHost)
```

Comment:
Would be nice to include the `err` here to provide more detail about the failure.

Reply:
Perhaps use structured logs (WithFields) as well for these;

```go
logrus.WithFields(logrus.Fields{logrus.ErrorKey: err, "host": realmHost}).Warn("TLS config not found for host")
```

Reply:
I think it is not a good idea to add an error log for this case, because it can be normal to not have a particular TLS config for the host of the challenge, per example in the standard case you mentionned above : if the auth tocken provider accept the same TLS config as the docker registry (in this case, the certificates are just specified in etc/docked/certs.d/<registry.domain>, not in /etc/docked/certs.d/<token_provider.domain>) 
If we add an error log here, in the standard case this log will happen but the connection on the token provider will be successfull.
The fix I did tries to change the TLS config for the auth challenge on the token provider only if a particular TLS config exist on the client for the token provider, if not it keep the same behaviour as before my fix.